cppcache/src/CacheImpl.cpp

```diff
@@ -88,14 +88,11 @@ CacheImpl::CacheImpl(Cache* c, const std::string& name,
     LOGINFO("Heap LRU eviction controller thread started");
   }
 
-  m_cacheStats = new CachePerfStats(m_distributedSystem.get()
-                                        ->getStatisticsManager()
-                                        ->getStatisticsFactory());
   m_expiryTaskManager->begin();
 
   m_initialized = true;
-
-  m_poolManager = std::unique_ptr<PoolManager>(new PoolManager(*m_implementee));
+  m_pdxTypeRegistry = std::make_shared<PdxTypeRegistry>(this);
```

Comment:
Is `this` not initialized enough in the initializer list to initialize `m_pdxTypeRegistry` there?

Reply:
I changed after update to pass CacheImpl instead of Cache.

`this` might be initialized enough, but I didn't want to rely on that in case PdxTypeRegistry later changed to require more of the CacheImpl object before being constructed.